src/driver.ml

```diff
@@ -323,6 +325,21 @@ module Transform = struct
         registered_at = Caller_id.get ~skip:[];
       }
 
+  (* Meant to be used after partitioning *)
+  let rewrites_not_context_free t =
+    match t with
+    | { name; _ } when String.equal name builtin_context_free_name -> false
```

Comment:
ppxlib's driver will assemble several AST traversal passes. One of them is meant to apply all context-free rules, it is built by ppxlib itself and is named "<builtin:context-free>".

All other passes come from ppx-es registering `str -> str` or `sig -> sig` rewriting functions.

Indeed there are no guarantees a user won't use that name when registering a ppx of their own but it is very unlikely.

I think this all part of the code could be refactored and simplified significantly and that might help distinguish between this special pass and all others.

In the meantime we could try to forbid this name for outside users. The transformation name is only used for error reporting as far as I remember so it usually is the name of the ppx itself, it's unlikely that we would break anything by adding such a restriction.

I think it should be alright but if you feel this is too much of a risk, I'm happy to look into refactoring this part in a way that allows us not to check by name.